platform/linuxbsd/display_server_x11.cpp

```diff
@@ -1453,6 +1453,32 @@ void DisplayServerX11::window_set_drop_files_callback(const Callable &p_callable
 	wd.drop_files_callback = p_callable;
 }
 
+bool DisplayServerX11::window_is_centered(WindowID p_window) {
+	_THREAD_SAFE_METHOD_
+
+	if (!windows.has(p_window)) {
+		return false;
+	}
+
+	Point2i screen_size = screen_get_size(0);
+	Point2i window_size = window_get_size(p_window);
+	Point2i position = screen_size / 2 - window_size / 2;
+
+	return (window_get_position(p_window) == position);
+}
+
+void DisplayServerX11::window_set_centered(WindowID p_window) {
+	_THREAD_SAFE_METHOD_
+
+	ERR_FAIL_COND(!windows.has(p_window));
+
+	Point2i screen_size = screen_get_size(0);
+	Point2i window_size = window_get_size(p_window);
+	Point2i position = screen_size / 2 - window_size / 2;
```

Comment:
```suggestion
	Point2i position = screen_get_position(0) + screen_size / 2 - window_size / 2;
```

Window position is in "virtual desktop" coordinates, which always starts at `(0, 0)`, but screen `0`, is not necessary at the `(0, 0)` coordinates, therefore screen position should be taken into account (same for all functions / platforms).

e.g., if you have screens configured like this, `(0, 0)` (marked with ❌) is outside any screen:

<img width="397" alt="Screenshot 2022-09-16 at 08 44 56" src="https://user-images.githubusercontent.com/7645683/190565715-5559437c-5226-4a1b-bdcf-2b20e8398044.png">